docarray/document/mixins/multimodal.py

```diff
@@ -0,0 +1,134 @@
+from dataclasses import is_dataclass
+
+import typing
+from enum import Enum
+
+from docarray.types import ImageDocument, BlobDocument, TextDocument
+
+if typing.TYPE_CHECKING:
+    from docarray import Document, DocumentArray
+
+
+class AttributeType(Enum):
+    DOCUMENT = 1
+    PRIMITIVE = 2
+    ITERABLE_PRIMITIVE = 3
+    ITERABLE_DOCUMENT = 4
+    NESTED = 5
+    ITERABLE_NESTED = 6
+
+
+class MultiModalMixin:
+    @classmethod
+    def from_dataclass(cls, obj):
```

Comment:
So the `from_dataclass` can also be applied to the `Document` class? Can we restrict this function to `MultiModalDocument` only ?